src/api/bulk2.ts

```diff
@@ -0,0 +1,1000 @@
+import { EventEmitter } from 'events';
+import { Duplex, Readable, Writable } from 'stream';
+import Connection from '../connection';
+import { Serializable, Parsable } from '../record-stream';
+import HttpApi from '../http-api';
+import { StreamPromise } from '../util/promise';
+import { registerModule } from '../jsforce';
+import { BulkRequest } from './bulk';
+import { getLogger, Logger } from '../util/logger';
+import { concatStreamsAsDuplex } from '../util/stream';
+import { HttpResponse, Record, Schema, Optional } from '../types';
+import is from '@sindresorhus/is';
+
+export type IngestOperation =
+  | 'insert'
+  | 'update'
+  | 'upsert'
+  | 'delete'
+  | 'hardDelete';
+
+export type QueryOperation = 'query' | 'queryAll';
+
+export type JobStateV2 =
+  | 'Open'
+  | 'UploadComplete'
+  | 'InProgress'
+  | 'JobComplete'
+  | 'Aborted'
+  | 'Failed';
+
+export type QueryJobInfoV2 = {
+  id: string;
+  operation: QueryOperation;
+  object: string;
+  createdById: string;
+  createdDate: string;
+  systemModstamp: string;
+  state: 'UploadComplete' | 'InProgress' | 'Aborted' | 'JobComplete' | 'Failed';
+  concurrencyMode: 'Parallel';
+  contentType: 'CSV';
+  apiVersion: string;
+  jobType: 'V2Query';
+  lineEnding: 'LF' | 'CRLF';
+  columnDelimiter:
+    | 'BACKQUOTE'
+    | 'CARET'
+    | 'COMMA'
+    | 'PIPE'
+    | 'SEMICOLON'
+    | 'TAB';
+  numberRecordsProcessed: string;
+  retries: string;
+  totalProcessingTime: string;
+  isPkChunkingSupported?: boolean;
+};
```

Comment:
this used to be one type for both query and ingest jobs, there's some overlap but for instance states are different and also have unique props.

query job info:
https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/query_get_one_job.htm

ingest job info:
https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/get_job_info.htm